c7n/resources/asg.py

```diff
@@ -492,6 +492,49 @@ def process(self, asgs, event=None):
         return super(VpcIdFilter, self).process(asgs)
 
 
+@filters.register('failing')
+class FailingFilter(Filter):
+    schema = type_schema(
+        'failing', count={'type': 'number'})
+
+    def asg_failed_launch(self, activities={}, token=None):
+        limit_date = datetime.now(tz=tzutc()) - timedelta(days=1)
```

Comment:
date should be configurable, by default scaling activities are across the last 6 weeks, a date comparison should be used to stop the pagination, else its just whatever gets you to 100.